pom.xml

```diff
@@ -0,0 +1,493 @@
+                <exclude>**/grok/patterns/**</exclude>
+                <exclude>conf/**</exclude>
+                <exclude>data/**</exclude>
+                <exclude>plugins/**</exclude>
+                <exlcude>**/*.patch</exlcude>
+                <exclude>**/logrotate.d/**</exclude>
+                <exclude>**/limits.d/**</exclude>
+                <exclude>**/*.json</exclude>
+                <exclude>**/*.json.template</exclude>
+                <exclude>**/MANIFEST.MF</exclude>
+                <!-- exclude any hack classes -->
+                <exclude>**/org/apache/hadoop/**</exclude>
+                <!-- exclude resource files-->
+                <exclude>**/resources/**</exclude>
+              </excludes>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>2.17</version>
+        <executions>
+          <execution>
+            <id>validate</id>
+            <phase>process-test-classes</phase>
+            <configuration>
+              <configLocation>checkstyle.xml</configLocation>
+              <suppressionsLocation>suppressions.xml</suppressionsLocation>
+              <encoding>UTF-8</encoding>
+              <consoleOutput>true</consoleOutput>
+              <failsOnError>true</failsOnError>
+              <includeTestSourceDirectory>true</includeTestSourceDirectory>
+              <excludes>**/org/apache/cassandra/**,**/org/apache/hadoop/**</excludes>
+            </configuration>
+            <goals>
+              <goal>check</goal>
+            </goals>
+          </execution>
+        </executions>
+        <dependencies>
+          <dependency>
+            <groupId>com.puppycrawl.tools</groupId>
+            <artifactId>checkstyle</artifactId>
+            <version>6.19</version>
+          </dependency>
+        </dependencies>
+      </plugin>
+      <plugin>
+        <groupId>io.cdap</groupId>
+        <artifactId>cdap-maven-plugin</artifactId>
+        <version>1.1.0</version>
+        <configuration>
+          <cdapArtifacts>
+            <parent>system:cdap-data-pipeline[6.0.0,7.0.0-SNAPSHOT)</parent>
+            <parent>system:cdap-data-streams[6.0.0,7.0.0-SNAPSHOT)</parent>
+          </cdapArtifacts>
+        </configuration>
+        <executions>
+          <execution>
+            <id>create-artifact-config</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>create-plugin-json</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <version>3.5.0</version>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <_exportcontents>
+              com.fasterxml.*;
```

Comment:
Only after adding Olingo V4 dependency. For example, `com.fasterxml.*;` was added to solve: 
```
Caused by: java.lang.RuntimeException: Provider for class javax.xml.stream.XMLInputFactory cannot be created
	at javax.xml.stream.FactoryFinder.findServiceProvider(FactoryFinder.java:367) ~[na:1.8.0_144]
	... 40 common frames omitted
Caused by: java.util.ServiceConfigurationError: javax.xml.stream.XMLInputFactory: Provider com.fasterxml.aalto.stax.InputFactoryImpl not found
	at java.util.ServiceLoader.fail(ServiceLoader.java:239) ~[na:1.8.0_144]
	at java.util.ServiceLoader.access$300(ServiceLoader.java:185) ~[na:1.8.0_144]
	at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:372) ~[na:1.8

```
I guess, I missed to remove this, because initially I started implementation with two clients, but decided to split into multiple PRs.

Reply:
can you remove the ones that aren't needed for this PR and introduce them in the later PR? I would want to see if it is caused by a specific method call and see if we could do some things to make it work without exporting.

Reply:
Ok. Removed already. Just forgot to mention that.